astropy/utils/data.py

```diff
@@ -56,7 +85,11 @@ class Conf(_config.ConfigNamespace):
         aliases=['astropy.coordinates.name_resolve.name_resolve_timeout'])
     compute_hash_block_size = _config.ConfigItem(
         2 ** 16,  # 64K
-        'Block size for computing MD5 file hashes.')
+        'Block size for computing file hashes.')
+    hash_algorithm = _config.ConfigItem(
+        'sha224',
```

Comment:
Why 224 instead of 256 or 512?

Also this probably should be discussed and resolved in https://github.com/astropy/astropy/issues/9303 before being implemented here.

Reply:
I initially picked 512 but the filenames became unpleasantly long. sha-224 has no known security holes (unlike md5, sha1), is always supported by hashlib, and seems to be the shortest hash with those properties. 

I can easily remove the code if the conclusion on #9303 is that this isn't worth it. But this demonstrates that the implementation is not difficult.